src/main/java/school/hei/haapi/endpoint/rest/security/SecurityConf.java

```diff
@@ -410,6 +410,10 @@ req, res, null, forbiddenWithRemoteInfo(req))))
                     .hasAnyRole(TEACHER.getRole(), MANAGER.getRole())
                     .requestMatchers(PUT, "/groups/*/awarded_courses")
                     .hasAnyRole(MANAGER.getRole())
+                    .requestMatchers(GET, "/teachers/*/awarded_courses")
+                    .hasAnyRole(STUDENT.getRole(), TEACHER.getRole(), MANAGER.getRole())
+                    .requestMatchers(PUT, "/teachers/*/awarded_courses")
+                    .hasAnyRole(MANAGER.getRole())
                     .requestMatchers(
```

Comment:
We have new pattern for the security conf ansure that those antpath is already in authenticated filter before